elitzur-core/src/main/scala/com/spotify/elitzur/validators/Validator.scala

```diff
@@ -244,6 +247,95 @@ object Validator extends Serializable {
     override def shouldValidate: Boolean = true
   }
 
+  //scalastyle:off method.length cyclomatic.complexity
+  /**
+   * Core validation loop for both dynamic and derived validators
+   * This code is deliberately mutable, uses casts to avoid unboxing, and avoids dereferencing
+   * Ignore Validators in order to optimize speed at runtime
+   */
+  @SuppressWarnings(Array("org.wartremover.warts.Var"))
+  def validationLoop[T](validatorAccessors: Array[ValidatorAccessor[Any]],
+                     constructor: Seq[Any] => T,
+                     outermostClassName: String,
+                     path: String = "",
+                     config: ValidationRecordConfig = DefaultRecordConfig)
+                    (implicit reporter: MetricsReporter): PostValidation[T] = {
+    val cs = new Array[Any](validatorAccessors.length)
+    var i = 0
+    var atLeastOneValid = false
+    var atLeastOneInvalid = false
+    val counterClassName = outermostClassName
+
+    while (i < validatorAccessors.length) {
```

Comment:
Maybe I'm missing something, but wouldn't it be easier to iterate via:

```scala
validatorAccessors.map {
    case accessor => {
            ...
        }
}
```

Reply:
This is a performance optimization. `while` with indexed array access outperforms `.map{}`

Since Elitzur as a library was primarily directly used in pipelines, people had a lot of concerns about SLOs and delays in delivery, so we optimize for performance over idiomatic scala

Reply:
👍 That makes sense, thanks for clarifying!